### PR TITLE
Use summarize_for_lesson_edit when returning resource/vocabulary search matches

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Completes [PLAT-736](https://codedotorg.atlassian.net/browse/PLAT-736). 

tldr: As Dani predicted, the wrong information was being passed down. It doesn't appear there was any data loss so this is mostly a cosmetic fix. We didn't get a bug report for vocabulary because it doesn't actually use properties as of right now, but I also fixed that spot.

Slightly longer version: the search was returning `{id: <id>, name: <name>, ... , properties: { type: <type>, audience: <audience> ...}}` when React was expecting a flat map. So type and audience weren't there from the client's perspective and showed up as blank.

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]

**Review by dmcavoy — APPROVED** [review]
Nice find Bethany!
